packages/repository/src/relations/has-one/has-one.repository.ts

```diff
@@ -39,6 +40,30 @@ export interface HasOneRepository<Target extends Entity> {
     filter?: Pick<Filter<Target>, Exclude<keyof Filter<Target>, 'where'>>,
     options?: Options,
   ): Promise<Target>;
+
+  /**
+   * Delete multiple target model instances
+   * @param where Instances within the where scope are deleted
+   * @param options
+   * @returns A promise which resolves the deleted target model instances
+   */
+  delete(where?: Where<Target>, options?: Options): Promise<Count>;
+
+  /**
+   * Patch related target model instance
+   * @param dataObject The target model fields and their new values to patch
+   * @param options
+   * @returns A promise which resolves the patched target model instances
+   */
+  patch(dataObject: DataObject<Target>, options?: Options): Promise<Count>;
+
+  /**
+   * Put related target model instance
+   * @param dataObject The target model fields and their new values to put
+   * @param options
+   * @returns A promise which resolves the patched target model instances
+   */
+  put(dataObject: DataObject<Target>, options?: Options): Promise<void>;
```

Comment:
As mentioned above, I am proposing to remove `put` from this pull request.

If we decide to keep it: please use the name `replace` for consistency with LB 3.x API. `replace` describes the intent better (replace all properties of a model instance), `put` is too REST/HTTP-specific.

Reply:
Part of my answer is above with the constrains to implement isNames() method due to calling existing instruction "this.definition.idName()" implemented into the abstract class.
But, as per your proposing, I am removing the PUT operation and these questions will be handled  in the new PR.